### PR TITLE
feat(planting-plant-flow): Garden & MaxingElixir stacks

### PR DESCRIPTION
**Pull request by emberglaz**
* add images
* add IPlant model
* add GardenStackNavigator
* add MixingElixirStackNavigator
* add updatePlantArea action in chaseSlice
* add GardenScreen UX, positioning & navigation
* add GardenTutorialDialogScreen UX, positioning & navigation
* add Plant component UX, positioning
* add PlantArea component UX, positioning
* add ElixirAnimationScreen UX, positioning & navigation
* add ElixirInstructionScreen UX, positioning & navigation
* add MixingElixirSuccessScreen UX, positioning & navigation
* remove ElixirAnimationScreen
* remove ElixirButtonScreen

### CODE REVIEW

File: app/models/IPlant.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { PlantAreaType } from '../screens/Garden/components';
+import { PlantsType } from '../utils/types';
+
+export interface IPlant {
+  image: keyof typeof PlantsType;
+  plantArea: keyof typeof PlantAreaType;
```
Comment (Reykjavik151) [inline app/models/IPlant.ts]: you can use enum directly: `image: PlantsType;`

File: app/navigation/stacks/mixingElixirStackNavigator.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  ElixirInstruction: {
+    phase: MixingElixirPhaseType;
+    isFirstTimeGarden: boolean;
+    plantImage: keyof typeof PlantsType;
+    selectedPlantArea: keyof typeof PlantAreaType;
+  };
+  MixingElixirSuccess: {
+    plantImage: keyof typeof PlantsType;
+    isFirstTimeGarden: boolean;
+    selectedPlantArea: keyof typeof PlantAreaType;
```
Comment (Reykjavik151) [inline app/navigation/stacks/mixingElixirStackNavigator.tsx]: the same - why do we need `keyof typeof` for enums?

File: app/redux/slices/cacheSlice.ts [hunk 3]
```diff
@@ -33,6 +36,13 @@ const INITIAL_STATE: ICacheState = {
   translations: null,
   trySomethingItem: null,
   nicknames: null,
+  plantArea: {
+    TopLeft: null,
+    TopRight: null,
+    Center: null,
+    BottomLeft: null,
+    BottomRight: null,
+  },
```
Comment (Reykjavik151) [inline app/redux/slices/cacheSlice.ts:53]: `TODO: make it as separate slice`

File: app/redux/slices/cacheSlice.ts [hunk 4]
```diff
@@ -62,6 +72,10 @@ export const cacheSlice = createSlice({
         state.nicknames.firstPlayer = payload;
       }
     },
+    updatePlantArea(state, { payload }: PayloadAction<IPlant>) {
+      state.plantArea[payload.plantArea] = payload;
+    },
```
Comment (Reykjavik151) [inline app/redux/slices/cacheSlice.ts:86]: The same `TODO:` comment

File: app/redux/types/cacheTypes.ts [hunk 5]
```diff
@@ -48,3 +52,5 @@ export interface ITrySomethingItemPayload {
 export interface INicknamesPayload {
   payload: INicknames;
 }
+
+export type IPlantArea = Record<keyof typeof PlantAreaType, Nullable<IPlant>>;
```
Comment (Reykjavik151) [inline app/redux/types/cacheTypes.ts]: the same -- we need use `enum` without `keyof typeof`

File: app/screens/Garden/Garden/Garden.tsx [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+  const [activePlantArea, setActivePlantArea] =
+    useState<Nullable<keyof typeof PlantAreaType>>(null);
+
+  useMount(() => {
+    if (isFirstTime) {
+      setTimeout(() => {
+        navigation.navigate('GardenTutorialDialog', { isStart: false });
+      }, 3000);
+    }
```
Comment (Reykjavik151) [inline app/screens/Garden/Garden/Garden.tsx:38]: Please clarify this logic for me

File: app/screens/Garden/Garden/Garden.tsx [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+
+export const GardenScreen: React.FC<IGardenScreenProps> = ({
+  navigation,
+  route,
+}) => {
+  const { isPlanting, isFirstTime, isFirstTimeGarden } = route.params;
+  const { t } = useTranslation();
+
+  const [activePlantArea, setActivePlantArea] =
+    useState<Nullable<keyof typeof PlantAreaType>>(null);
```
Comment (Reykjavik151) [inline app/screens/Garden/Garden/Garden.tsx]: the same as above

File: app/screens/Garden/GardenTutorialDialog/GardenTutorialDialog.tsx [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+import { DialogView } from '../../../components';
+import { GARDEN_TUTORIAL_DIALOG } from './GardenTutorialDialog.data';
+import { IGardenTutorialDialogScreenProps } from './GardenTutorialDialog.types';
+
+export const GardenTutorialDialogScreen: React.FC<IGardenTutorialDialogScreenProps> =
+  ({ navigation, route }) => {
+    const { isStart } = route.params;
+
+    const onSabmit = useCallback(() => {
```
Comment (Reykjavik151) [inline app/screens/Garden/GardenTutorialDialog/GardenTutorialDialog.tsx:11]: `onsAbmit` typing mistake

File: app/screens/Garden/GardenTutorialDialog/GardenTutorialDialog.tsx [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+        onSubmit={onSabmit}
+        navigateBetween={
+          isStart
+            ? {
+                index: 6,
+                onPress: goToGarden,
+              }
+            : null
+        }
+        initialIdx={isStart ? 0 : 7}
```
Comment (Reykjavik151) [inline app/screens/Garden/GardenTutorialDialog/GardenTutorialDialog.tsx:42]: make indexes with constants

File: app/screens/Garden/components/Plant/Plant.styles.ts [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+  plant: ImageStyle;
+}
+
+export const styles = StyleSheet.create<IPlantStyles>({
+  plant: {
+    position: 'absolute',
+    height: 132,
+    width: 90,
+    transform: [{ rotate: '45deg' }],
+    zIndex: 1000,
```
Comment (Reykjavik151) [inline app/screens/Garden/components/Plant/Plant.styles.ts:13]: use just `100`

File: app/screens/Garden/components/Plant/Plant.tsx [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+
+  useMount(() => {
+    const nowSeconds = moment().format('X');
+    const growthSeconds = +nowSeconds - +plant.plantedAt;
+    setCorrectPlantImage(growthSeconds);
+  });
+
+  return (
+    <Image
+      source={plantImage ? plantImage : IMAGES.LOGO}
```
Comment (Reykjavik151) [inline app/screens/Garden/components/Plant/Plant.tsx:42]: plantImage ?? IMAGES.LOGO

File: app/screens/Garden/components/PlantArea/PlantArea.tsx [hunk 12]
```diff
@@ -0,0 +1,186 @@
...
+              style={[
+                styles.plantArea,
+                styles.plantAreaBorder,
+                activePlantArea === PlantAreaType.BottomLeft &&
+                  styles.activePlantArea,
+              ]}
+              disabled={!!plantArea?.BottomLeft}
+              onPress={() => selectPlantArea(PlantAreaType.BottomLeft)}>
+              {plantArea?.BottomLeft && (
+                <Plant
```
Comment (Reykjavik151) [inline app/screens/Garden/components/PlantArea/PlantArea.tsx]: Refactor it with `render...` function + params

File: app/screens/Garden/components/PlantArea/PlantArea.types.ts [hunk 13]
```diff
@@ -0,0 +1,15 @@
+import { Nullable } from '../../../../utils';
+
+export interface IPlantAreaProps {
+  isPlanting: boolean;
+  activePlantArea: Nullable<keyof typeof PlantAreaType>;
+  setActivePlantArea: (value: Nullable<keyof typeof PlantAreaType>) => void;
```
Comment (Reykjavik151) [inline app/screens/Garden/components/PlantArea/PlantArea.types.ts]: the same as above

File: app/screens/MixingElixir/ElixirAnimation/ElixirAnimation.tsx [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+    switch (phase) {
+      case MixingElixirPhaseType.Mix: {
+        return 'screens.mixing_exixir.elixir_animaion.mix';
+      }
+      case MixingElixirPhaseType.Open: {
+        return 'screens.mixing_exixir.elixir_animaion.open';
+      }
+      default: {
+        return 'screens.mixing_exixir.elixir_animaion.pour';
+      }
```
Comment (Reykjavik151) [inline app/screens/MixingElixir/ElixirAnimation/ElixirAnimation.tsx]: we don't need to wrap each `case` with `{}`

File: app/screens/MixingElixir/ElixirAnimation/ElixirAnimation.tsx [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+          selectedPlantArea,
+          isFirstTimeGarden,
+        });
+      } else {
+        navigation.push('MixingElixirSuccess', {
+          plantImage,
+          selectedPlantArea,
+          isFirstTimeGarden,
+        });
+      }
```
Comment (Reykjavik151) [inline app/screens/MixingElixir/ElixirAnimation/ElixirAnimation.tsx]: if (...) {
logic 1
return
}

logic 2

File: app/screens/MixingElixir/ElixirAnimation/ElixirAnimation.tsx [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+
+export const ElixirAnimationScreen: React.FC<IElixirAnimationScreenProps> = ({
+  navigation,
+  route,
+}) => {
+  const { phase, plantImage, selectedPlantArea, isFirstTimeGarden } =
+    route.params;
+  const { t } = useTranslation();
+
+  const getCorrectTitleKey = useCallback(() => {
```
Comment (Reykjavik151) [inline app/screens/MixingElixir/ElixirAnimation/ElixirAnimation.tsx]: refactor it with `useMemo`

File: app/screens/MixingElixir/ElixirInsrtuction/ElixirInstruction.tsx [hunk 17]
```diff
@@ -0,0 +1,62 @@
...
+        case MixingElixirPhaseType.Open: {
+          setTitleKey('screens.mixing_exixir.elixir_instruction.open');
+          setButtonTitleKey('buttons.tap_to_open');
+          break;
+        }
+        default: {
+          setTitleKey('screens.mixing_exixir.elixir_instruction.pour');
+          setButtonTitleKey('buttons.tap_to_pour');
+          break;
+        }
```
Comment (Reykjavik151) [inline app/screens/MixingElixir/ElixirInsrtuction/ElixirInstruction.tsx]: Refactor `case` without `{}`

File: app/screens/MixingElixir/MixingElixirSuccess/MixingElixirSuccess.tsx [hunk 18]
```diff
@@ -0,0 +1,65 @@
...
+
+      dispatch(
+        cacheSlice.actions.updatePlantArea({
+          plantArea: selectedPlantArea,
+          image: plantImage,
+          plantedAt: nowSeconds,
+        }),
+      );
+      navigation.push('GardenStack', {
+        screen: 'Garden',
```
Comment (Reykjavik151) [inline app/screens/MixingElixir/MixingElixirSuccess/MixingElixirSuccess.tsx]: spaceline after `dispatch`

File: app/utils/types.ts [hunk 19]
```diff
@@ -14,3 +14,13 @@ export interface ITranslations {
...
 }
+
+export enum MixingElixirPhaseType {
+  Mix = 1,
+  Open = 2,
+  Pour = 3,
+}
+
+export enum PlantsType {
+  SPOUR_COMPASSION = 'SPOUR_COMPASSION',
```
Comment (Reykjavik151) [inline app/utils/types.ts]: SpourCompassion = 'SPOUR_COMPASSION'

**Review by Reykjavik151 — CHANGES_REQUESTED** [review]

File: app/screens/MixingElixir/ElixirAnimation/ElixirAnimation.tsx (hunk 15 above)
Comment (emberglaz) [reply app/screens/MixingElixir/ElixirAnimation/ElixirAnimation.tsx]: 🐍

File: app/screens/Garden/components/PlantArea/PlantArea.styles.ts [hunk 20]
```diff
@@ -0,0 +1,86 @@
...
+  emptyPlantArea: ViewStyle;
+}
+
+export const styles = StyleSheet.create<IPlantingPlantStyles>({
+  container: {
+    height: 450,
+    width: 350,
+    alignSelf: 'center',
+    zIndex: 0,
+    backgroundColor: 'red',
```
Comment (Reykjavik151) [inline app/screens/Garden/components/PlantArea/PlantArea.styles.ts]: remove

File: app/screens/Garden/components/PlantArea/PlantArea.styles.ts [hunk 21]
```diff
@@ -0,0 +1,86 @@
...
+  },
+  tree: {
+    position: 'absolute',
+    resizeMode: 'contain',
+    alignSelf: 'center',
+    height: WINDOW_HEIGHT < 600 ? 278 / 1.5 : 278,
+    width: WINDOW_HEIGHT < 600 ? 212 / 1.5 : 212,
```
Comment (Reykjavik151) [inline app/screens/Garden/components/PlantArea/PlantArea.styles.ts]: WINDOW_HEIGHT < 600 to variable

File: app/screens/Garden/components/PlantArea/PlantArea.styles.ts [hunk 22]
```diff
@@ -0,0 +1,86 @@
...
+  },
+  plantArea: {
+    transform: isIOS
+      ? [{ skewX: '30deg' }]
+      : [{ rotateX: '50deg' }, { rotateY: '30deg' }, { rotateZ: '-30deg' }],
+    height: isIOS ? 90 : 120,
+    width: isIOS ? 90 : 120,
+    zIndex: 10,
+    top: WINDOW_HEIGHT < 600 ? -85 : 0,
+    left: WINDOW_HEIGHT < 600 ? 55 : 0,
```
Comment (Reykjavik151) [inline app/screens/Garden/components/PlantArea/PlantArea.styles.ts]: the same as above

File: app/screens/Garden/components/PlantArea/PlantArea.tsx [hunk 23]
```diff
@@ -0,0 +1,111 @@
...
+      currentPlantArea: PlantAreaType,
+      currentPlant: IPlant | null,
+      additionalStyles?: ViewStyle[],
+    ) => {
+      if (isPlanting) {
+        return (
+          <Pressable
+            style={[
+              styles.plantArea,
+              additionalStyles && [...additionalStyles],
```
Comment (Reykjavik151) [inline app/screens/Garden/components/PlantArea/PlantArea.tsx]: `additionalStyles,`

File: app/screens/Garden/components/PlantArea/PlantArea.tsx [hunk 24]
```diff
@@ -0,0 +1,111 @@
...
+      setActivePlantArea(selectedPlace);
+    },
+    [setActivePlantArea],
+  );
+
+  const renderPlantArea = useCallback(
+    (
+      currentPlantArea: PlantAreaType,
+      currentPlant: IPlant | null,
+      additionalStyles?: ViewStyle[],
```
Comment (Reykjavik151) [inline app/screens/Garden/components/PlantArea/PlantArea.tsx]: ViewStyle[] | ViewStyle

File: app/screens/Garden/components/PlantArea/PlantArea.tsx [hunk 25]
```diff
@@ -0,0 +1,111 @@
...
+          </Pressable>
+        );
+      }
+
+      return (
+        <View
+          style={[
+            styles.plantArea,
+            styles.borderTransparent,
+            additionalStyles && [...additionalStyles],
```
Comment (Reykjavik151) [inline app/screens/Garden/components/PlantArea/PlantArea.tsx]: the same as above

File: app/screens/Garden/components/PlantArea/PlantArea.tsx [hunk 26]
```diff
@@ -0,0 +1,111 @@
...
+          ])}
+        </View>
+        <View style={generalStyles.row}>
+          <View style={styles.emptyPlantArea} />
+
+          {/* //* BottomRight plant area */}
+          {renderPlantArea(PlantAreaType.BottomRight, plantArea?.BottomRight, [
+            styles.plantAreaBorder,
+            styles.plantAreaBottom,
+          ])}
```
Comment (Reykjavik151) [inline app/screens/Garden/components/PlantArea/PlantArea.tsx]: the same as above

**Review by Reykjavik151 — CHANGES_REQUESTED** [review]

**Review by Reykjavik151 — APPROVED** [review]

**Pull request merged**